ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OrphanFilesCleanService.java

```diff
@@ -107,107 +108,106 @@ public void run() {
+      clearInternalTableMetadata(keyedArcticTable, keyedArcticTable.baseTable(), lastTime);
+      clearInternalTableMetadata(keyedArcticTable, keyedArcticTable.changeTable(), lastTime);
     } else {
-      clearInternalTableDataFiles(table, internalTable, lastTime, execute, exclude);
+      clearInternalTableMetadata(arcticTable, arcticTable.asUnkeyedTable(), lastTime);
     }
   }
 
-  private static void clearInternalTableDataFiles(ArcticTable table, UnkeyedTable internalTable, long lastTime,
-                                                  boolean execute, Set<String> exclude) {
+  private static void clearInternalTableContentsFiles(ArcticTable table, UnkeyedTable internalTable, long lastTime,
+                                                      Set<String> exclude) {
     int deleteFilesCnt = 0;
     String dataLocation = internalTable.location() + File.separator + DATA_FOLDER_NAME;
     if (table.io().exists(dataLocation)) {
       for (FileStatus fileStatus : table.io().list(dataLocation)) {
         deleteFilesCnt += deleteInvalidDataFiles(table.io(),
             fileStatus,
             lastTime,
-            exclude,
-            execute);
+            exclude);
       }
     }
-    LOG.info("{} total delete[execute={}] {} files", table.id(), execute, deleteFilesCnt);
+    LOG.info("{} total delete {} files", table.id(), deleteFilesCnt);
   }
 
-  private static void clearInternalTableMetadata(ArcticTable table, UnkeyedTable internalTable, long lastTime,
-                                                 boolean execute) {
+  private static void clearInternalTableMetadata(ArcticTable table, UnkeyedTable internalTable, long lastTime) {
     Set<String> validFiles = getValidMetadataFiles(table.id(), table.io(), internalTable);
     LOG.info("{} table get {} valid files", table.id(), validFiles.size());
+    Pattern excludeFileNameRegex = getExcludeFileNameRegex(internalTable);
+    LOG.info("{} table get exclude file name pattern {}", table.id(), excludeFileNameRegex);
     int deleteFilesCnt = 0;
     String metadataLocation = internalTable.location() + File.separator + METADATA_FOLDER_NAME;
     LOG.info("start orphan files clean in {}", metadataLocation);
     for (FileStatus fileStatus : table.io().list(metadataLocation)) {
       deleteFilesCnt += deleteInvalidMetadata(table.io(),
           fileStatus,
-          validFiles,
           lastTime,
-          execute);
+          validFiles,
+          excludeFileNameRegex);
     }
-    LOG.info("{} total delete[execute={}] {} manifestList/manifest/metadata files", table.id(), execute,
+    LOG.info("{} total delete {} manifestList/manifest/metadata files", table.id(),
         deleteFilesCnt);
   }
 
+  private static Pattern getExcludeFileNameRegex(UnkeyedTable table) {
+    String latestFlinkJobId = null;
+    for (Snapshot snapshot : table.snapshots()) {
+      String flinkJobId = snapshot.summary().get("flink.job-id");
+      if (StringUtils.isNotBlank(flinkJobId)) {
```

Comment:
I recommend using `com.google.common.base.Strings#isNullOrEmpty`.

Reply:
OK, I changed it to `!Strings.isNullOrEmpty(flinkJobId)`.